rotkehlchen/globaldb/handler.py

```diff
@@ -1031,6 +1031,9 @@ def edit_user_asset(asset: AssetWithOracles) -> None:
                 f'Tried to edit non existing asset with identifier {asset.identifier}',
             ) from e
 
+        if asset.is_evm_token():
+            GlobalDBHandler().edit_evm_token(cast(EvmToken, asset))
```

Comment:
shouldnt''t you return or something here? Otherwise this will go further down with `asset.is_crypto()` evaluating to `True` and the asset would be edited there too.

Also just lose the cast and use #type: ignore saying its' checked in the if.

Reply:
Yes, will make sense to return here. Thanks